core/trie/proof.go

```diff
@@ -139,19 +141,20 @@ func transformNode(tri *Trie, parentKey *Key, sNode StorageNode) (*Edge, *Binary
 func pathSplitOccurredCheck(mergedPath []ProofNode, nodeHashes map[felt.Felt]ProofNode) error {
 	splitHappened := false
 	for _, node := range mergedPath {
-		if node.Edge != nil {
+		switch node := node.(type) {
+		case *Edge:
 			continue
-		}
-
-		_, leftExists := nodeHashes[*node.Binary.LeftHash]
-		_, rightExists := nodeHashes[*node.Binary.RightHash]
-
-		if leftExists && rightExists {
-			if splitHappened {
-				return errors.New("split happened more than once")
+		case *Binary:
+			_, leftExists := nodeHashes[*node.LeftHash]
+			_, rightExists := nodeHashes[*node.RightHash]
+			if leftExists && rightExists {
+				if splitHappened {
+					return errors.New("split happened more than once")
+				}
+				splitHappened = true
 			}
-
-			splitHappened = true
+		default:
+			return ErrUnknownProofNode
```

Comment:
can we pass what type it was? i.e. something like that `fmt.Errorf("%w: %T", ErrUnknownProofNode, node)`